apps/include/apps.h

```diff
@@ -45,6 +45,8 @@
  */
 # define _UC(c) ((unsigned char)(c))
 
+# define IS_IP_ADDR(host) ((host) != NULL && ((*(host) >= '0' && *(host) <= '9') || *(host) == '['))
```

Comment:
Hmm. So according to this `7-zip.org` is an IP address!

Reply:
That's deeply wrong. To determine whether a string is an IP address, one needs to pass  to `getaddrinfo(3)` with AI_NUMERICHOST among the hints flags and `AF_UNSPEC` as the address family. This will cover both v4 and v6 addressed. Names enclosed in `[]` are presumptively address literals since OpenSSL does not support these around hostnames (in MTAs these often disable MX lookups, but that's not applicable here).

Reply:
Fixed, introducing instead a proper `APP_is_IP_address()` function along the lines given by Viktor.